src/jit/hwintrinsicxarch.cpp

```diff
@@ -1030,6 +1032,119 @@ GenTree* Compiler::impSSE2Intrinsic(NamedIntrinsic        intrinsic,
             break;
         }
 
+        case NI_SSE2_SetAllVector128:
+        {
+            assert(sig->numArgs == 1);
+            baseType = getBaseTypeOfSIMDType(sig->retTypeSigClass);
+            assert((baseType >= TYP_BYTE) && (baseType <= TYP_DOUBLE));
+            assert(baseType != TYP_FLOAT);
+
+            op1          = impPopStack().val;
+            GenTree* src = nullptr;
+
+            switch (baseType)
+            {
+                case TYP_DOUBLE:
+                {
+                    src     = gtNewSimdHWIntrinsicNode(TYP_SIMD16, op1, NI_SSE2_SetScalarVector128, baseType, simdSize);
+                    retNode = gtNewSimdHWIntrinsicNode(TYP_SIMD16, src, gtNewIconNode(0b01000100), NI_SSE2_Shuffle,
```

Comment:
Should this not be `gtNewIconNode(0)` with a base type of `TYP_DOUBLE` which will select bits `0:63` for all elements?

Reply:
Actually, `UnpackLow(op1, op1)` looks to be slightly better codegen.

Reply:
This is 3 op instruction I would need to clone src. This explicitly avoids cloning.